### PR TITLE
refine Ready Platform to get InternalIp for some testing

### PR DESCRIPTION
**Pull request by harz566**

### CODE REVIEW

File: TestProviders/ReadyProvider.psm1 [hunk 1]
```diff
@@ -26,6 +26,28 @@
 ###############################################################################################
 using Module ".\TestProvider.psm1"
 
+Function Get-IPAddressFromIfconfig([string] $ifConfigInfo) {
+	Write-Host "GetIPAddress $ifConfigInfo"
```
Comment (LiliDeng) [inline TestProviders/ReadyProvider.psm1]: Write-LogDbg "GetInternalIPAddress $ifConfigInfo"

File: TestProviders/ReadyProvider.psm1 [hunk 2]
```diff
@@ -26,6 +26,28 @@
...
+
+Function Set-InternalIPAddress([object] $AllVMData) {
+	$count=0
+	foreach ($vmData in $AllVMData) {
+		$ifconfigInfo = Run-LinuxCmd -username $global:user -password $global:password -ip $($vmData.PublicIp) -port $($vmData.SSHPort) -command "ifconfig"
+		$ipAddress = Get-IPAddressFromIfconfig -ifConfigInfo $ifConfigInfo
+		if ($ipAddress) {
+			$AllVmData[$count].InternalIP = $ipAddress
+		}
+		else {
```
Comment (LiliDeng) [inline TestProviders/ReadyProvider.psm1]: } else {

File: TestProviders/ReadyProvider.psm1 [hunk 3]
```diff
@@ -26,6 +26,28 @@
...
 
+Function Get-IPAddressFromIfconfig([string] $ifConfigInfo) {
+	Write-Host "GetIPAddress $ifConfigInfo"
+	[regex] $re = "(?:[0-9]{1,3}\.){3}[0-9]{1,3}"
+	[string] $matchedIp = $re.Match($ifConfigInfo)
+	return $matchedIp
+}
+
+Function Set-InternalIPAddress([object] $AllVMData) {
+	$count=0
```
Comment (LiliDeng) [inline TestProviders/ReadyProvider.psm1]: $count = 0

File: TestProviders/ReadyProvider.psm1 (hunk 1 above)
Comment (harz566) [reply TestProviders/ReadyProvider.psm1]: updated

File: TestProviders/ReadyProvider.psm1 (hunk 2 above)
Comment (harz566) [reply TestProviders/ReadyProvider.psm1]: updated

File: TestProviders/ReadyProvider.psm1 (hunk 3 above)
Comment (harz566) [reply TestProviders/ReadyProvider.psm1]: updated

File: TestProviders/ReadyProvider.psm1 [hunk 4]
```diff
@@ -28,6 +28,27 @@ using Module ".\TestProvider.psm1"
...
+		Write-LogDbg "GetIPAddress $ifConfigInfo"
+		[regex] $re = "(?:[0-9]{1,3}\.){3}[0-9]{1,3}"
+		[string] $matchedIp = $re.Match($ifConfigInfo)
+		return $matchedIp
+	}
+
+	[void] SetInternalIPAddress([object] $AllVMData) {
+		$count = 0
+		foreach ($vmData in $AllVMData) {
+			$ifconfigInfo = Run-LinuxCmd -username $global:user -password $global:password -ip $($vmData.PublicIp) -port $($vmData.SSHPort) -command "ifconfig"
```
Comment (juhlee-microsoft) [inline TestProviders/ReadyProvider.psm1:41]: 1) We used 'ip addr show' as ifconfig will be depreciated.
2) This should be with runAsSudo

File: TestProviders/ReadyProvider.psm1 [hunk 5]
```diff
@@ -28,6 +28,27 @@ using Module ".\TestProvider.psm1"
...
+			$ipAddress = GetIPAddressFromIfconfig -ifConfigInfo $ifConfigInfo
+			if ($ipAddress) {
+				$AllVmData[$count].InternalIP = $ipAddress
+			} else {
+				Write-LogErr "Cannot get the internal IP address for $($vmData.PublicIp):$($vmData.SSHPort)"
```
Comment (juhlee-microsoft) [inline TestProviders/ReadyProvider.psm1:46]: Consistent manner of terminology use: Private IP Address

**Review by juhlee-microsoft — CHANGES_REQUESTED** [review]

**Comment by juhlee-microsoft** [comment]
@harz566 Please add the context in the description.

**Comment by juhlee-microsoft** [comment]
@harz566 when you have time back, please reopen this PR.

**Pull request closed without merging**